dist/ExtUtils-ParseXS/lib/ExtUtils/ParseXS/Node.pm

```diff
@@ -0,0 +1,113 @@
+package ExtUtils::ParseXS::Node;
+use strict;
+use warnings;
+
+our $VERSION = '3.55';
+
+=head1 NAME
+
+ExtUtils::ParseXS::Node - Classes for nodes of an ExtUtils::ParseXS AST
+
+=head1 SYNOPSIS
+
+XXX TBC
+
+=head1 DESCRIPTION
+
+XXX Sept 2024: this is Work In Progress. This API is currently private and
+subject to change. Most of ParseXS doesn't use an AST, and instead
+maintains just enough state to emit code as it parses. This module
+represents the start of an effort to make it use an AST instead.
+
+An C<ExtUtils::ParseXS::Node> class, and its various subclasses, hold the
+state for the nodes of an Abstract Syntax Tree (AST), which represents the
+parsed state of an XS file.
+
+Each node is basically a hash of fields. Which field names are legal
+varies by the node type. The hash keys and values can be accessed
+directly: there are no getter/setter methods.
+
+=cut
+
+
+# ======================================================================
+
+package ExtUtils::ParseXS::Node;
+
+# Base class for all the other node types.
+#
+# The 'use fields' enables compile-time or run-time errors if code
+# attempts to use a key which isn't listed here.
+
+my $USING_FIELDS;
+
+BEGIN {
+    our @FIELDS = (
+        # Currently there are no node fields common to all node types
+    );
+
+    # do 'use fields', except: fields needs Hash::Util which is XS, which
+    # needs us. So only 'use fields' on systems where Hash::Util has already
+    # been built.
+    if (eval 'require Hash::Util; 1;') {
+        require 'fields.pm';
```

Comment:
Why not `require fields;` ?

Reply:
Because I was cargo-culting it from similar code I added a few weeks ago in ParseXS.pm. That original code had 'fields.pm' probably as a by-product of my several attempts to get fields to work or be bypassed in a miniperl-ish environment. They are essentially the same thing: "require X" is converted at compile time into "require 'X.pm'", so it doesn't really matter. But I suppose aesthetically "require fields" is better.